### PR TITLE
Check namespace when follow tag reference

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
PROVISION CLOUD RESOURCES
/retest

**Comment by bparees** [comment]
my understanding is it's not just cross-namespace references that are supposed to be invalid.  it's any cross-imagestream reference, even within the same namespace.

**Comment by dmage** [comment]
Yes, we already have test for it: "crosss image tag reference error".

File: pkg/image/apis/image/helper.go [hunk 1]
```diff
@@ -414,6 +414,10 @@ func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *
 			return tag, &tagRef, multiple, nil
 		}
 
+		if tagRef.From.Namespace != "" && tagRef.From.Namespace != stream.ObjectMeta.Namespace {
+			return tag, nil, multiple, ErrCrossImageStreamReference
```
Comment (miminar) [inline pkg/image/apis/image/helper.go]: Could you fix the docstring of the error as well?

Comment (dmage) [reply pkg/image/apis/image/helper.go]: Nice catch, fixed.

**Comment by bparees** [comment]
/lgtm

but i still don't think this resolves the issues in https://bugzilla.redhat.com/show_bug.cgi?id=1530340

**Pull request merged**